manifests/tar_create.pp

```diff
@@ -15,6 +15,6 @@
   exec { "generate ${path}":
     cwd     => '/root',
     path    => ['/usr/bin', '/bin'],
-    command => "tar -czf ${path} ssl-build/*.noarch.rpm ssl-build/${foreman_proxy_fqdn}/*.noarch.rpm",
+    command => "tar -czf ${path} ssl-build/*.noarch.rpm ssl-build/*.crt ssl-build/${foreman_proxy_fqdn}/*",
```

Comment:
Previously we are generating a tarball with `ssl-build/*.noarch.rpm` and `ssl-build/${foreman_proxy_fqdn}/*.noarch.rpm`

Now we still get `ssl-build/*.noarch.rpm`, and additionally `ssl-build/*.crt`.

And for the proxy, change to the wildcard path `ssl-build/${foreman_proxy_fqdn}/*`

It's clear from reading the description that this is intended to be an intermediate design that allows to gradually move towards the intended design -- is this why you still collect `ssl-build/*.noarch.rpm` if it exists, and additionally `ssl-build/*.crt`?

Why collect those separately, but for the proxy fqdn, a wildcard without any trailing characters? Wouldn't it be better to be much more specific about what we intend to find there?

Reply:
Also perhaps this is overkill, but in trying to review this with more limited knowledge of the current workings, something like this might be nice:

```puppet
# collect the existing RPM for backwards compatibility, to be removed in the future.
$rpm_rel_path = "ssl-build/*.noarch.rpm"

# also collect the .crt file. added with puppet-certs #352
$crt_file_rel_path = "ssl-build/*.crt"

# now all files in the $foreman_proxy_fqdn subdir , not only RPMs
$proxy_fqdn_wildcard_path = "ssl-build/${foreman_proxy_fqdn}/*"
```

Then build the command with:

```puppet
command => "tar -czf ${path} ${rpm_rel_path} ${crt_file_rel_path} ${proxy_fqdn_wildcard_path}",
```

Obviously this would work the same, but we (myself in particular) might appreciate this in the future when we revisit to make the next phase of changes.

Reply:
> What is less clear to me: why collect the above separately and somewhat specifically, but for the proxy fqdn, a wildcard without any trailing characters? Wouldn't it be better to be much more specific about what we intend to find there?

We do not want to collect all of `ssl-build/` because there is openssl configuration, CA key, serial etc. in that directory. For the directories that are for the proxy at, e.g. `ssl-build/${hostname}` we want the certificate and keys and the RPMs, so it just seemed easier to grab everything and not accidentally miss something since we effectively need it all right now.

Reply:
I have updated with these ideas in mind.

Reply:
Thanks! It's a lot clearer to me now and I'm sure I'll appreciate it later when we have to revisit this again for the next steps.

Reply:
Indeed, this naming makes it quite clear. Thanks!